engine/src/flutter/shell/platform/embedder/embedder.h

```diff
@@ -258,6 +258,40 @@ typedef enum {
   kFlutterSemanticsFlagIsRequired = 1 << 30,
 } FlutterSemanticsFlag;
 
+typedef struct {
```

Comment:
This will need a `struct_size` member. See: https://github.com/flutter/flutter/blob/b8aa0fd8468bc419d53b95345721d28e332a8212/engine/src/flutter/shell/platform/embedder/embedder.h#L12-L50

Reply:
Does that mean we can never change the size of the struct even if we have more flags later?

@chunhtai  If you have to take a guess, how many semantics flags will we eventually need?

Reply:
No you can add more members in the future, but embedders will use `struct_size` to check whether it can access that new member.

Reply:
still, if we can't change the size of the struct, we need to predict how many semantics flags we need in the future.

Reply:
`struct_size` is always populated with `sizeof(FlutterSemanticsFlags)`, so it will increase as fields are added. 

This is a bit related to why fields must always be added at the end of a struct and why we can never reorder fields. Embedders built against older versions of the engine will be reading these fields from the memory offsets within the struct from the engine version they built against. If the embedder author drops in a new engine shared lib, those offsets must still match correctly.

Reply:
thanks for explaining this!